pages/learn/develop/container-contracts.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Container contracts
+excerpt: A guide to using container contracts to enforce device compatibility
+---
+
+# Container contracts
+
+__Note__ Container contracts are available for devices running Supervisor versions >= 10.6.17. All prior versions will not enforce any contracts.
+
+Container contracts are used to ensure the compatibility of a device to run an individual service of an application. For example, container contracts may be used to ensure that the device is running a specific version of [{{ $names.company.lower }} Supervisor][supervisor] or has a specific version of the [NVIDIA Tegra Linux Driver Package][l4t] (L4T).
+
+Each service container can define the contracts that it requires. If a container's contract requirements are not met, the release is not deployed to the device, unless it is marked as [optional](#optional-containers). You only need to define a contract for service containers that you wish to enforce a contract upon. Currently, you may define contract requirements based on the versions of Supervisor and L4T.
+
+## Contract specification
+
+Container contracts are defined as a `contract.yml` file and must be placed in the root of the build context for a service. For example, the following [multi-container application][multicontainer] defines two services named `first-service` and `second-service`, each with a corresponding `contract.yml` file.
+
+```shell
+.
+├── docker-compose.yml
+├── first-service
+│ ├── contract.yml
+│ └── Dockerfile.template
+└── second-service
+ ├── contract.yml
+ └── Dockerfile.template
+```
+
+`contract.yml` is a YAML file containing `type`, `slug`, `name`, and `requires` keys, which define the requirements of the contract. The following example requires the device to be running a version of L4T equal to 28.2 for the contract to pass. For L4T exact versions (either 28.2 or 32.2) must be specified.
+
+```yaml
+type: 'sw.container'
+slug: 'enforce-l4t-version'
+name: 'Enforce L4T Version'
+requires:
+ - type: 'sw.l4t'
+ version: '28.2'
+```
+
+* `type` should be set to `sw.container` for container contracts.
+* `slug` should be a unique identifier to identify the contract.
+* `name` should be a descriptive name of the contract.
+* `requires` should be a list of requirements to be enforced by the contract defined by `type` and `version` keys. Currently, `type` may be either `sw.supervisor` for the device Supervisor version or `sw.l4t` for the L4T version.
+
+__Note__ If `name`, `type`, or `slug` are omitted, the build will fail.
+
+Multiple requirements may be added to a single contract. The following example requires that the Supervisor version is greater than or equal to 10.6.13, and the L4T version is equal to 32.2.
+
+```yaml
+type: 'sw.container'
+slug: 'enforce-supervisor-and-l4t'
+name: 'Enforce supervisor and l4t requirements'
+requires:
+ - type: 'sw.supervisor'
+ version: '>=10.6.17'
```

Comment:
I believe version needs to be at the same indentation as the `type:` string, otherwise this yaml would fail to be parsed

Reply:
Thanks for the catch. Corrected indentation.